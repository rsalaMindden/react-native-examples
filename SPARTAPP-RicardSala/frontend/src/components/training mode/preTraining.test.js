import React from "react";
import PreTraining from "./preTraining";
import { render,fireEvent }from '../../utils/tetsUtils/test.utils'


const navigation={
    push:jest.fn()
};
   
    const route={
        params:{
            training:[{name: 'FOR TIME',
                exercises: [{
                  exercise: { name: 'BURPEES',
                    description: 'asdasd',
                    picture: 'asd',
                    done:false,
                    },
                  reps: 10,
                }],
              }]}
        };


describe('Given a PreTraining component',()=>{
    describe('When it is render',()=>{
        let screen;
        beforeEach(()=>{
            screen=render(<PreTraining navigation={navigation} route={route}></PreTraining>);
        });
        test('Then should match tHe snapshot',()=>{
            expect(screen).toMatchSnapshot();
          });
          describe('And AMRAP button is clicked', () => {
            test('Then navigation.push should have been called with handleClickAMRAP', () => {
              const AMRAP = screen.getByTestId('HomePage');
              fireEvent.press(AMRAP);
              expect(navigation.push).toHaveBeenCalled();
            });
          });
      
        });
    });
 