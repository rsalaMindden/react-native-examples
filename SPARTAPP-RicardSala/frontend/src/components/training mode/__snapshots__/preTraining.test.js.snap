// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a PreTraining component When it is render Then should match tHe snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#595260",
        "flex": 1,
      },
      Object {
        "alignItems": "center",
      },
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/spartappYELLOW.png",
        }
      }
      style={
        Object {
          "height": 31,
          "marginTop": 54,
          "width": 146,
        }
      }
      testID="HomePage"
    />
  </View>
  <RCTScrollView
    horizontal={false}
  >
    <View>
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#2C2E43",
              "borderRadius": 15,
              "display": "flex",
              "elevation": 15,
              "flexDirection": "row",
              "flexWrap": "wrap",
              "justifyContent": "center",
              "marginBottom": 25,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 7,
                "width": 0,
              },
              "shadowOpacity": 0.43,
              "shadowRadius": 9.51,
              "width": 300,
            },
            Object {
              "alignItems": "center",
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flexDirection": "row",
              "height": 150,
              "justifyContent": "center",
              "margin": 5,
              "width": 180,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#FFD523",
                },
                Object {
                  "fontSize": 20,
                  "fontWeight": "600",
                  "justifyContent": "center",
                  "marginLeft": 10,
                  "marginRight": 10,
                },
              ]
            }
          >
            BURPEES
          </Text>
          <Image
            source={
              Object {
                "uri": "asd",
              }
            }
            style={
              Object {
                "height": 150,
                "marginLeft": 20,
                "resizeMode": "contain",
                "width": 80,
              }
            }
          />
        </View>
        <View
          style={
            Array [
              Object {},
              Object {
                "textAlign": "center",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#FFD523",
                },
                Object {
                  "margin": 10,
                },
              ]
            }
          >
            asdasd
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
